### ISSUE TITLE
Regression on openssl binary with des-ede3

### ISSUE THREAD

**Issue by juergenchrist** [openssl/openssl#17587]
`des-ede3` cipher now needs an iv on the command line.  With openssl 1.x:
```
# openssl des-ede3 -e -K 938caafe6b5a0fbd0151687e -in enc -out res
hex string is too short, padding with zero bytes to length
# echo $?
0
```
With openssl 3.x:
```
# openssl des-ede3 -e -K 938caafe6b5a0fbd0151687e -in enc -out res
iv undefined
# echo $?
1
```
I see in providers/implementations/ciphers/cipher_tdes.c that this is on purpose to prevent code breakage.  Unfortunately this now leads to a breakage for clients using the command line tool.

**Comment by ifranzki** [comment]
In OpenSSL 1.1.1 the DES_EDE3 cipher is constructed in evp/e_des3.c:

```
    BLOCK_CIPHER_defs(des_ede3, DES_EDE_KEY, NID_des_ede3, 8, 24, 8, 64,
                  EVP_CIPH_RAND_KEY | EVP_CIPH_FLAG_DEFAULT_ASN1,
                  des_ede3_init_key, NULL, NULL, NULL, des3_ctrl)
```

where the 64 is the IV length.
However, the ECB mode define is the following:

```
#define BLOCK_CIPHER_defs(cname, kstruct, \
                          nid, block_size, key_len, iv_len, cbits, flags, \
                          init_key, cleanup, set_asn1, get_asn1, ctrl) \
....
BLOCK_CIPHER_def_ecb(cname, kstruct, nid, block_size, key_len, flags, \
                     init_key, cleanup, set_asn1, get_asn1, ctrl)
```
It does not pass the iv_len to BLOCK_CIPHER_def_ecb, and BLOCK_CIPHER_def_ecb sets 0 as IV length:

```
#define BLOCK_CIPHER_def_ecb(cname, kstruct, nid, block_size, key_len, \
                             flags, init_key, cleanup, set_asn1, \
                             get_asn1, ctrl) \
BLOCK_CIPHER_def1(cname, ecb, ecb, ECB, kstruct, nid, block_size, key_len, \
                  0, flags, init_key, cleanup, set_asn1, get_asn1, ctrl)
```
This sets the 0 into EVP_CIPHER as iv_len.

Consequently EVP_CIPHER_iv_length() will return 0 for an ECB mode cipher:
```
int EVP_CIPHER_iv_length(const EVP_CIPHER *cipher)
{
    return cipher->iv_len;
}
```

In OpenSSL 3.0 EVP_CIPHER_iv_length is an alias of EVP_CIPHER_get_iv_length.

So clearly, in OpenSSL 3.0 EVP_CIPHER_iv_length  behaves differently than in 1.1.1.

**Comment by t8m** [comment]
Fix in #17591
